OPAL/si/src/main/scala/org/opalj/fpcf/scheduling/MaximumPhaseScheduling.scala

```diff
@@ -97,38 +102,59 @@ abstract class MultiplePhaseScheduling extends SchedulingStrategy {
             computePhase(ps, phaseAnalyses, remainingAnalyses)
         }
 
-        if(config.getBoolean(DisableCleanup)) calculateDeletions(schedule, ps, config) else schedule
+        if (!config.getBoolean(DisableCleanup)) calculateDeletions(schedule, ps, config) else schedule
     }
 
-    private def calculateDeletions[A](schedule: List[PhaseConfiguration[A]], ps: PropertyStore, config: Config): List[PhaseConfiguration[A]] = {
-        def getSetOrNilForProperty(propertyKey: String) = {
-            if (!config.hasPath(propertyKey)) Nil
-            else {
-                val rawValue = config.getString(propertyKey).trim
-                if (rawValue.isEmpty) Set.empty[Int]
-                else rawValue.split(",").map(_.trim.toInt).toSet
+    private def calculateDeletions[A](
+        schedule: List[PhaseConfiguration[A]],
+        ps:       PropertyStore,
+        config:   Config
+    ): List[PhaseConfiguration[A]] = {
+        def keyByName(name: String): SomePropertyKey = {
+            val max = PropertyKey.maxId
+            var i = 0
+            while (i <= max) {
+                if (PropertyKey.name(i) == name) return PropertyKey.key(i)
+                i += 1
             }
+            throw new IllegalArgumentException(s"Unknown property name: $name")
+        }
+        def getSetForProperty(propertyKey: String): Set[Int] = {
+            Option(
+                config.getString(propertyKey)
+            ).toSeq.flatMap(_.split(",")).filter(_.nonEmpty).map(name => keyByName(name).id).toSet
         }
-        val pkToKeep = getSetOrNilForProperty(PropertiesToKeepKey)
-        val pkToDelete = getSetOrNilForProperty(PropertiesToRemoveKey)
-         val properties: mutable.Set[Int] =
-            schedule.iterator.flatMap(_.propertyKinds.propertyKindsComputedInThisPhase.map(key => key.id)).to(mutable.Set)
+        val pkToKeep = getSetForProperty(PropertiesToKeepKey)
```

Comment:
Still unsure whether this should really be solved via the config. It seems too variable to me. Also, it doesn't allow for a simple programmatic way to set these.